tensorboard/plugins/graph/tf_graph_common/minimap.ts

```diff
@@ -155,7 +155,23 @@ export class Minimap {
     let $download = d3.select('#graphdownload');
     this.download = <HTMLLinkElement>$download.node();
     $download.on('click', d => {
-      this.download.href = this.downloadCanvas.toDataURL('image/png');
+      // Revoke the old URL, if any. Then, generate a new URL.
+      URL.revokeObjectURL(this.download.href);
+      // We can't use the `HTMLCanvasElement.toBlob` API because it does
+      // not have a synchronous variant, and we need to update this href
+      // synchronously. Instead, we create a blob manually from the data
+      // URL.
+      const dataUrl = this.downloadCanvas.toDataURL("image/png");
+      const prefix = dataUrl.slice(0, dataUrl.indexOf(","));
+      if (!prefix.endsWith(":base64") && !prefix.endsWith(";base64")) {
+        console.warn(`non-base64 data URL (${prefix}); cannot use blob download`);
+        this.download.href = dataUrl;
+        return;
+      }
+      const data = atob(dataUrl.slice(dataUrl.indexOf(",") + 1));
+      const bytes = new Uint8Array(Array.from(data).map(c => c.charCodeAt(0)));
```

Comment:
Instead of creating a single-chart string for each element of data, can we just change the index to charCodeAt()?  E.g. what this does: https://stackoverflow.com/a/21797381/1179226

It seems almost guaranteed to be cheaper to simply index the string at different points versus sub-slicing the string at then indexing into the sub-slice.

Reply:
Great.  Nice to have those numbers since then I guess toBlob isn't much advantage anyway.